kernel/kernel-keymanager-service/src/main/java/io/mosip/kernel/clientcrypto/service/impl/LocalClientCryptoServiceImpl.java

```diff
@@ -232,8 +230,28 @@ private boolean doesKeysExists() {
         return (keysDir.exists() && Objects.requireNonNull(keysDir.list()).length >= 2);
     }
 
+    //Copy ${user.home}/.mosipkeys/db.conf to ${user.dir}/.mosipkeys/db.conf
+    private void backwardCompatibilityFix() {
+        Path targetPrivateKey = Paths.get(ClientCryptoManagerConstant.KEY_PATH, ClientCryptoManagerConstant.KEYS_DIR, PRIVATE_KEY);
+        Path sourcePrivateKey = Paths.get(System.getProperty("user.home"), ClientCryptoManagerConstant.KEYS_DIR, PRIVATE_KEY);
+        if(!targetPrivateKey.toFile().exists() && sourcePrivateKey.toFile().exists()) {
```

Comment:
Can we make this simple? If(targetPrivateKey.toFile().exists()) { //Then no need to copy, just return.
}
Continue the remaining code

Reply:
changed as per the comment sasi